fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/security/KeyManager.java

```diff
@@ -22,46 +26,136 @@
 import java.security.GeneralSecurityException;
 import java.util.Arrays;
 import java.util.Base64;
+import java.util.Comparator;
+import java.util.Optional;
 
 import javax.inject.Inject;
 import javax.inject.Singleton;
 
+/** Service manging keys and providing easy access to the main device key and PoP Tokens */
 @Singleton
 public class KeyManager {
```

Comment:
the API of this class confuses me a bit, between `retrieveValidPoPToken`, `getPoPToken` and `requirePoPToken`, they all sound sort of similar.

1. Do we need both `requirePoPToken` and `getPoPToken`, or could you drop one ? The only (likely) failure scenario is the wallet not being initialized. I also don't see them being used in the code.

2. Is there a distinction between `retrieve` and `get` in this context? It might be easier to quickly distinguish between `getPoPToken` and `getValidPoPToken`

3. In my understanding, both `get` and `require` are actually generating a PoP token, which would simplify the API to just `getValidPoPToken` and `generatePoPToken`.

Reply:
I think I will keep `require` and `get`.
Require throws Exceptions when something goes wrong as the PoPToken is required where Get optionally return the token.